week-4/Homework/mandatory/4-groceries.js

```diff
@@ -28,19 +28,30 @@ Exercise 1:
   Then use console.log() to print out the list.
 */
 // Gather all week item names into this array
+
 let weeklyGroceriesToBuy = [];
+for (let property in weeklyMealPlan[property]) {
+  weeklyGroceriesToBuy.push(weeklyMealPlan);
+}
 
+console.log(weeklyGroceriesToBuy.flat());
 
+console.log(weeklyGroceriesToBuy);
 
 /*
 Exercise 2:
-  Loop through your list again, but now only collect the weekend items into the weeklyGroceriesToBuy array.
+  Loop through your list again, but now only collect the weekend items into the weekendGroceriesToBuy array.
   Then use console.log() to print out the list.
 */
 // Gather weekend item names into this array
-let weekendGroceriesToBuy = [];
-
+// let weekendGroceriesToBuy = [];
 
+for (let property in weeklyMealPlan) {
+  if (property.startsWith("s")) {
+    weekendGroceriesToBuy.push(weeklyMealPlan[property]);
+  }
+}
+console.log(weekendGroceriesToBuy.flat());
```

Comment:
You should have put the names of the weekends and look if the object contains those names of weekends, have a look at the video loop through an object.